### PR TITLE
5 2pt setup reducer for fetching doctors

### PR DESCRIPTION
**Pull request by muneebulrehman**
action/reducers for fetching a single doctor and a list of doctors

### CODE REVIEW

File: src/features/doctor/doctorSlice.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    builder.addCase(fetchDoctors.pending, (state) => {
+      state.loading = true;
+    });
+    builder.addCase(fetchDoctors.fulfilled, (state, action) => {
+      state.doctors = action.payload;
+      state.loading = false;
+      state.error = '';
+    });
+    builder.addCase(fetchDoctors.rejected, (state, action) => {
+      state.error = action.error.message;
```
Comment (damdafayton) [inline src/features/doctor/doctorSlice.js:37]: Maybe double check this? I am not sure if error message is handled on api.

File: src/App.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import { BrowserRouter, Routes } from 'react-router-dom';
 import './App.css';
+import DoctorView from './features/doctor/DoctorView';
 
 function App() {
   return (
     <BrowserRouter>
       <div className="App">
         <h1>React app</h1>
+        <DoctorView />
```
Comment (damdafayton) [inline src/App.js:11]: Maybe its better to set this inside Route for '/' ?

**Review by damdafayton — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 2 above)
Comment (muneebulrehman) [reply src/App.js:11]: It was just a dummy file to check whether the action/reducers were working.

File: src/features/doctor/doctorSlice.js (hunk 1 above)
Comment (muneebulrehman) [reply src/features/doctor/doctorSlice.js:37]: Yeah, that is right. Not handling the errors.

**Pull request merged**